go/parquet/pqarrow/encode_arrow.go

```diff
@@ -449,7 +457,7 @@ func writeDenseArrow(ctx *arrowWriteContext, cw file.ColumnChunkWriter, leafArr
 				data[i] = parquet.ByteArray(valueBuf[offsets[i]:offsets[i+1]])
 			}
 		default:
-			return xerrors.New(fmt.Sprintf("invalid column type to write to ByteArray: %s", leafArr.DataType().Name()))
+			return xerrors.Errorf("invalid column type to write to ByteArray: %s", leafArr.DataType().Name())
```

Comment:
if we're in here changing this, might as well use `fmt.Errorf` as `xerrors` is no longer needed and should be phased out. do something like:

`return fmt.Errorf("%w: expected column type %s to be compatible with `ByteArray`", arrow.ErrInvalid, leafArr.DataType())`